src/promptflow-evals/promptflow/evals/evaluators/chat/__init__.py

```diff
@@ -0,0 +1,189 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+__path__ = __import__("pkgutil").extend_path(__path__, __name__)  # type: ignore
+
+from promptflow.entities import AzureOpenAIConnection
+from promptflow.evals.evaluators import GroundednessEvaluator, RelevanceEvaluator, CoherenceEvaluator, FluencyEvaluator
+from typing import List, Dict
+from concurrent.futures import ThreadPoolExecutor, as_completed
+import json
+import logging
+import numpy as np
+
+
+logger = logging.getLogger(__name__)
+
+
+class ChatEvaluator:
+    def __init__(self, model_config: AzureOpenAIConnection, deployment_name: str):
+        """
+        Initialize an evaluator configured for a specific Azure OpenAI model.
+
+        :param model_config: Configuration for the Azure OpenAI model.
+        :type model_config: AzureOpenAIConnection
+        :param deployment_name: Deployment to be used which has Azure OpenAI model.
+        :type deployment_name: AzureOpenAIConnection
+        :return: A function that evaluates and generates metrics for "chat" scenario.
+        :rtype: function
+
+        **Usage**
+
+        .. code-block:: python
+
+            eval_fn = ChatEvaluator(model_config, deployment_name="gpt-4")
+            conversation = [
+                {"role": "user", "content": "What is the value of 2 + 2?"},
+                {"role": "assistant", "content": "2 + 2 = 4", "context": {
+                    "citations": [{"id": "math_doc.md", "content": "Information about additions: 1 + 2 = 3, 2 + 2 = 4"}]}}
+            ]
+            result = chat_eval(conversation=conversation)
+        """
+
+        # TODO: Need a built-in evaluator for retrieval. It needs to be added to `self._rag_evaluators` collection
+        self._rag_evaluators  = [
+            GroundednessEvaluator(model_config, deployment_name=deployment_name),
+            RelevanceEvaluator(model_config, deployment_name=deployment_name),
+        ]
+        self._non_rag_evaluators = [
+            CoherenceEvaluator(model_config, deployment_name=deployment_name),
+            FluencyEvaluator(model_config, deployment_name=deployment_name),
+        ]
+
+    def __call__(self, *, conversation: List[Dict], parallel=True, **kwargs):
+        """Evaluates chat scenario.
+
+        :param conversation: The conversation to be evaluated. Each turn should have "role" and "content" keys.
+                             "context" key is optional for assistant's turn and should have "citations" key with list of citations.
```

Comment:
For evaluation like groundedness, relevance, and retrieval_score, "context" key is a must for assistant's role.